### DISCUSSION TITLE
Send multiple bytes with tuh_hid_set_report

### DISCUSSION THREAD

**Discussion by MarcoKonrad** [hathach/tinyusb#2095] [Q&A]
Hello everyone,

i have a question regarding the tuh_hid_set_report() function.
I do have a USB-Host-Device that has to recieve multiple bytes as a wakeup. 
For example it would recieve 0x020403 as wakeup request. I was able to get a keyboard to recieve the report, but keyboards only recieve a single Byte of Data, so the function looks like this:

tuh_hid_set_report(dev_addr,instance,0,HID_REPORT_TYPE_OUTPUT,&status2,1)

But if dont understand how to send multiple bytes simultaneously. My idea was to do it like this:

static unsigned char sendReport [3] = {0x02, 0x04, 0x03}; 
tuh_hid_set_report(dev_addr,instance,0,HID_REPORT_TYPE_OUTPUT,&sendReport,sizeof(sendReport));

But using a debugger i can see that in xfer the report only contains 0x02.

Has anyone an idea how to achieve this?

The Raspberry PI Pico is acting as the USB Host btw.

**Comment by Commotari** [comment]
did you specify the report in the descriptor to have the length 3?
(btw. i am not an expert on usb programming, just did some device programming and had only problems, if the length did not fit the descriptor, so, i thought that might be a similar problem in your case)

**Reply by MarcoKonrad** [reply]
Yes i did. I tried to hardcode the length and i tried it with sizeof(sendReport). Still get no wakeup on the Device Side (A LED should turn on in the case it recieves a USB Send). If i plug it into Windows and use other programs it works without problems.

**Comment by hathach** [comment]
try to enable tinyusb log may help to analyze the issue.
